@jethrogb I think it already errors when both features are selected?

```
cargo build --features parking_lot,spin
   Compiling namedlock v0.5.0 (file:///Users/messense/Projects/namedlock-rs)
error[E0252]: the name `Mutex` is defined multiple times
   --> src/lib.rs:119:49
    |
118 | #[cfg(feature="spin")] use spin::{Mutex,MutexGuard};
    |                                   ----- previous import of the type `Mutex` here
119 | #[cfg(feature="parking_lot")] use parking_lot::{Mutex,MutexGuard};
    |                                                 ^^^^^ `Mutex` reimported here
    |
    = note: `Mutex` must be defined only once in the type namespace of this module

error[E0252]: the name `MutexGuard` is defined multiple times
   --> src/lib.rs:119:55
    |
118 | #[cfg(feature="spin")] use spin::{Mutex,MutexGuard};
    |                                         ---------- previous import of the type `MutexGuard` here
119 | #[cfg(feature="parking_lot")] use parking_lot::{Mutex,MutexGuard};
    |                                                       ^^^^^^^^^^ `MutexGuard` reimported here
    |
    = note: `MutexGuard` must be defined only once in the type namespace of this module

error[E0252]: the name `Mutex` is defined multiple times
  --> src/ownedmutexguard.rs:61:49
   |
60 | #[cfg(feature="spin")] use spin::{Mutex,MutexGuard};
   |                                   ----- previous import of the type `Mutex` here
61 | #[cfg(feature="parking_lot")] use parking_lot::{Mutex,MutexGuard};
   |                                                 ^^^^^ `Mutex` reimported here
   |
   = note: `Mutex` must be defined only once in the type namespace of this module

error[E0252]: the name `MutexGuard` is defined multiple times
  --> src/ownedmutexguard.rs:61:55
   |
60 | #[cfg(feature="spin")] use spin::{Mutex,MutexGuard};
   |                                         ---------- previous import of the type `MutexGuard` here
61 | #[cfg(feature="parking_lot")] use parking_lot::{Mutex,MutexGuard};
   |                                                       ^^^^^^^^^^ `MutexGuard` reimported here
   |
   = note: `MutexGuard` must be defined only once in the type namespace of this module

warning: unused imports: `MutexGuard`, `Mutex`
   --> src/lib.rs:119:49
    |
119 | #[cfg(feature="parking_lot")] use parking_lot::{Mutex,MutexGuard};
    |                                                 ^^^^^ ^^^^^^^^^^
    |
    = note: #[warn(unused_imports)] on by default

warning: unused imports: `MutexGuard`, `Mutex`
  --> src/ownedmutexguard.rs:61:49
   |
61 | #[cfg(feature="parking_lot")] use parking_lot::{Mutex,MutexGuard};
   |                                                 ^^^^^ ^^^^^^^^^^

error: aborting due to 4 previous errors

error: Could not compile `namedlock`.

To learn more, run the command again with --verbose.
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -104,6 +104,9 @@
 
 #[cfg(all(test,not(feature="std")))] #[macro_use] extern crate std;
 
+#[cfg(all(feature="spin",feature="parking_lot"))]
+compile_error!("`spin` and `parking_lot` features can't be used together");
+
 #[cfg(feature="spin")] extern crate spin;
 #[cfg(feature="parking_lot")] extern crate parking_lot;
 #[cfg(feature="std")] extern crate core;

